core/src/main/java/io/grpc/NameResolverProvider.java

```diff
@@ -49,6 +49,8 @@
  */
 @Internal
 public abstract class NameResolverProvider extends NameResolver.Factory {
+  /** Only visible to be used in ManagedChannelImpl. Do not use. */
+  public static final String NO_PROVIDERS_SCHEME = "noproviders";
```

Comment:
I sort of liked how it didn't break the factory if you weren't aware of it, just `newNameResolver()` would return `null`.